Rocket.Chat/API/Clients/MessagesClient.swift

```diff
@@ -122,7 +122,21 @@ struct MessagesClient: APIClient {
 
         api.fetch(DeleteMessageRequest(roomId: message.rid, msgId: id, asUser: asUser)) { response in
             switch response {
-            case .resource: break
+            case .resource(let resource):
+                guard let resourceMessage = resource.message else {
+                    return Alert.defaultError.present()
+                }
+
+                let shouldKeepMessageLocally = AuthManager.isAuthenticated()?.settings?.messageShowDeletedStatus ?? true
+                if !shouldKeepMessageLocally {
+                    Realm.executeOnMainThread(realm: realm) { realm in
+                        realm.delete(message)
+                    }
+                }
+
+                if let unmanagedMessage = resourceMessage.unmanaged {
```

Comment:
Since the API doesn't return the type of the message, what do you think we set it manually = 'd' in case success = true ?

Reply:
Good catch! Just pushed the changes